src/main/kotlin/interactor/GetLowestAveragePricesForFruitsAndVegetablesInteractor.kt

```diff
@@ -9,25 +9,27 @@ class GetLowestAveragePricesForFruitsAndVegetablesInteractor (
         return  data
             .getAllCitiesData()
             .filter(::excludeNullSalariesAndLowQualityDataAndNullFruitsAndVegetablesPrices)
-            .sortedBy {getRatio(it)}
-            .map { it.cityName }
+            .sortedBy(::getPricesToSalaryAverage)
             .take(limit)
+            .map { it.cityName }
+
```

Comment:
```suggestion
    fun execute(limit: Int): List<String> {
        if (limit < 0) throw InvalidLimitException("Limit cannot be zero or negative")
        return  data
            .getAllCitiesData()
            .ifEmpty { throw IllegalStateException("Something went wrong") }
            .filter(::excludeNullSalariesAndLowQualityDataAndNullFruitsAndVegetablesPrices)
            .sortedBy(::getPricesToSalaryAverage)
            .take(limit)
            .map { it.cityName }
    }
```
Then change the test cases.